### PR TITLE
fix: broken tests after app init

### LINKED ISSUE #161: Fix: Tests are Failing after Init
With removal of the nexus-prisma-plugin several CRUD operations and default Input types, resolvers, etc are no longer created and thus the tests are not functioning as expected

Recreate:
`npx create-bison-app@1.10.8 my-test-app`
`yarn db:setup`
`yarn test`

Note tsConfig is deprecated and the video shared also shows the tests failing because of a bad import statement for server of the GraphQL setup as are the setup commands that need to be updated. This is resolved in PR #157  
After resolving those you will still have many failing tests

https://www.loom.com/share/4fa6d576fb0e4572a7372b698a40cded

### PR DESCRIPTION
**Pull request by cmejet**
This PR works to resolve the failing tests after the initialization of an app. Two test suites skipped that are failing on next/link components.

## Changes

- Add missing queries for User & Users that include authorization for all Users, resolves user tests
- Adding missing Inputs for User Create and Where clauses
- Skips the Login Form and Signup Form component tests as they have a failure on the dependent library that still needs to be resolved

## Screenshots

image

## Checklist

- [ ] Requires dependency update?
- [x] Generating a new app works

Fixes #161

### CODE REVIEW

File: packages/create-bison-app/template/graphql/modules/user.ts.ejs [hunk 1]
```diff
@@ -53,6 +53,27 @@ export const UserQueries = extendType({
       description: 'Returns the currently logged in user',
       resolve: (_root, _args, ctx) => ctx.user,
     });
+
+    // List Users Query (admin only)
+    t.field('users', {
+      type: list('User'),
+      resolve: async (_root, _args, ctx) => {
+        if (!isAdmin(ctx.user)) throw new ForbiddenError('Unauthorized');
```
Comment (cullylarson) [inline packages/create-bison-app/template/graphql/modules/user.ts.ejs:61]: I don't know enough about nexus to know if this is the best way, but nexus has an `authorize` function you can provide for auth checks. For example:

```js
      authorize: (_root, _args, ctx) => isAdmin(ctx.user),
```

File: packages/create-bison-app/template/graphql/modules/user.ts.ejs [hunk 2]
```diff
@@ -53,6 +53,27 @@ export const UserQueries = extendType({
...
+      },
+    });
+
+    // User Query
+    t.field('user', {
+      type: 'User',
+      args: {
+        where: nonNull(arg({ type: 'UserWhereUniqueInput' })),
+      },
+      resolve: async (_root, _args, ctx) => {
```
Comment (kgajera) [inline packages/create-bison-app/template/graphql/modules/user.ts.ejs:73]: I think we prefix the parameters with an underscore only if it's unused so the linter won't complain about it. If that's true, `_args` should be changed to `args` in this case and also in the `users` query.

File: packages/create-bison-app/template/jest.config.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ module.exports = {
   testPathIgnorePatterns,
   globals: {
     'ts-jest': {
-      tsConfig: {
+      tsconfig: {
```
Comment (cball) [inline packages/create-bison-app/template/jest.config.js:19]: 👍

File: packages/create-bison-app/template/tests/requests/user/createUser.test.ts [hunk 4]
```diff
@@ -46,7 +46,7 @@ describe.skip('User createUser mutation', () => {
       const admin = await UserFactory.create({ roles: { set: [Role.ADMIN] } });
 
       const variables = {
-        data: { email: 'hello@wee.net', password: 'fake', roles: { set: [Role.ADMIN] } },
+        data: { email: 'hello@wee.net', password: 'fake', roles: [Role.ADMIN] },
```
Comment (cball) [inline packages/create-bison-app/template/tests/requests/user/createUser.test.ts:49]: 👍

**Review by cball — CHANGES_REQUESTED** [review]
Looking good! Agree with comments from @kgajera and @cullylarson

File: packages/create-bison-app/template/graphql/modules/user.ts.ejs (hunk 1 above)
Comment (cmejet) [reply packages/create-bison-app/template/graphql/modules/user.ts.ejs:61]: You are absolutely right, I did that in one of the other functions and I didn't here :doh:

**Review by kgajera — APPROVED** [review]

**Pull request merged**